### PR TITLE
Issue #614: Resolve MethodWithTooManyParameters, Naming, Javadoc IDEA violations

### PR DESCRIPTION
**Pull request by Vyom-Yadav**
#614
Renamed `run(..)` -> `runPostGenerationAndPublication(..)`

### CODE REVIEW

File: releasenotes-builder/src/main/java/com/github/checkstyle/github/NotesBuilder.java [hunk 1]
```diff
@@ -129,7 +129,7 @@ public static Result buildResult(String localRepoPath, String authToken, String
      * @param commit commit information.
      * @param commitMessage commit message.
      * @throws IOException if an I/O error occurs.
-     * @throws GitAPIException if an error occurs when accessing Git API.
+     * @noinspection MethodWithTooManyParameters
```
Comment (rnveach) [inline releasenotes-builder/src/main/java/com/github/checkstyle/github/NotesBuilder.java:132]: https://github.com/checkstyle/checkstyle/issues/11277 ?

Comment (Vyom-Yadav) [reply releasenotes-builder/src/main/java/com/github/checkstyle/github/NotesBuilder.java:132]: Added `@noinspectionreason`

File: releasenotes-builder/src/main/java/com/github/checkstyle/github/NotesBuilder.java [hunk 2]
```diff
@@ -129,7 +129,9 @@ public static Result buildResult(String localRepoPath, String authToken, String
      * @param commit commit information.
      * @param commitMessage commit message.
      * @throws IOException if an I/O error occurs.
-     * @throws GitAPIException if an error occurs when accessing Git API.
```
Comment (nick-mancuso) [inline releasenotes-builder/src/main/java/com/github/checkstyle/github/NotesBuilder.java:132]: Please remove this in minor commit, since this is not related to changes here.

**Review by nick-mancuso — CHANGES_REQUESTED** [review]
Item:

File: releasenotes-builder/src/main/java/com/github/checkstyle/github/NotesBuilder.java (hunk 2 above)
Comment (Vyom-Yadav) [reply releasenotes-builder/src/main/java/com/github/checkstyle/github/NotesBuilder.java:132]: Commit mentions Javadoc problems too, this falls under `Javadoc declaration problems inspection`

Comment (Vyom-Yadav) [reply releasenotes-builder/src/main/java/com/github/checkstyle/github/NotesBuilder.java:132]:
```
Reports Javadoc comments and tags with the following problems:

- invalid tag names
- incomplete tag descriptions
- duplicated tags
- missing Javadoc descriptions

```

Comment (nick-mancuso) [reply releasenotes-builder/src/main/java/com/github/checkstyle/github/NotesBuilder.java:132]: Thanks for the explanation :)

**Review by nick-mancuso — APPROVED** [review]

**Review by rnveach — APPROVED** [review]
